content/articles/create-an-ajax-type-ahead-feature-for-a-website/index.md

```diff
@@ -0,0 +1,467 @@
+  background-repeat: no-repeat;
+}
+
+.header {
+  align-self: flex-start;
+  color: #00ffff;
+}
+
+.main {
+  width: 100%;
+  max-width: 600px;
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+}
+
+form {
+  padding: 0.3rem 0.8rem;
+  display: flex;
+  gap: 1rem;
+  align-items: center;
+  border: 1px solid #f0f8ff;
+  border-radius: 50px;
+}
+
+.ri-search-line {
+  font-size: 2rem;
+}
+
+input[type="search"] {
+  border: 0;
+  outline: none;
+  width: 90%;
+  color: #f0f8ff;
+  font-family: inherit;
+  font-size: 2rem;
+  background-color: transparent;
+}
+
+::placeholder {
+  color: #c2a00a;
+}
+
+.card {
+  max-height: 80vh;
+  min-width: 350px;
+  list-style: none;
+  overflow-block: scroll;
+  background-color: #f0f8ff;
+  color: #000000;
+  border-radius: 10px;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 1rem;
+}
+
+.card-item {
+  width: 100%;
+  position: relative;
+  padding: 1.5rem;
+  border-radius: 15px;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+}
+
+.card-item::after {
+  content: "";
+  width: 100%;
+  height: 1px;
+  background-color: saddlebrown;
+  position: absolute;
+  left: 0;
+  bottom: 0;
+}
+
+.hl {
+  color: red;
+}
+```
+
+You can style the CSS file according to your preference.
+
+### The Javascript
+### Getting the data
+The data for this project is from the [cities.json](https://gist.githubusercontent.com/Miserlou/c5cd8364bf9b2420bb29/raw/2bf258763cdddd704f8ffd3ea9a3e81d25e2c6f6/cities.json) file, a file that has information of states cities in America.
+
+From the file, we get access to the following information of each state in America:
+1. State name
+2. City name
+3. Population
+4. Latitude and longitude
+5. Growth_from_2000_to_2013
+6. Rank
+
+You can choose the data to display from the list above.
+
+- Get the search word and the `ul` from the UI using the code below:
+
+```js
+const searchText = document.querySelector('input[type="search"]');
+const card = document.querySelector(".card");
+```
+
+- Create the data endpoint and create an empty array as shown below:
+
+```js
+const dataEndpoint =
+"https://gist.githubusercontent.com/Miserlou/c5cd8364bf9b2420bb29/raw/2bf258763cdddd704f8ffd3ea9a3e81d25e2c6f6/cities.json";
+
+const cities = [];
+```
+
+- Now, we need to fetch our data from the endpoint and assign the returned value to the empty array we created.
+
+```js
+fetch(dataEndpoint)
+  .then((blobdata) => blobdata.json())
+  .then((data) => cities.push(...data));
+```
+
+> The blobdata object, in turn, does not contain the actual JSON response body but is instead a representation of the entire HTTP response. To extract the JSON body content from the Response object, we use the `JSON()` method, which returns a second promise that resolves with the result of parsing the response body text as JSON. -[MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)
+
+![Blobdata output](./returned-blobdata.png)
```

Comment:
```suggestion
![Blobdata output](/engineering-education/create-an-ajax-type-ahead-feature-for-a-website/returned-blobdata.png)
```
Use the same format for your other images as well. Also, if your images are from third party sources/ are not screenshots, add the image source links.